packages/pwa/app/utils/routes-utils.test.js

```diff
@@ -7,38 +7,223 @@
 import React from 'react'
 
 import {configureRoutes} from './routes-utils'
-import {getUrlConfig} from './utils'
+import {getConfig} from './utils'
 
 jest.mock('./utils', () => {
     const original = jest.requireActual('./utils')
     return {
         ...original,
-        getUrlConfig: jest.fn()
+        getConfig: jest.fn()
     }
 })
 
 describe('configureRoutes', function() {
-    test('should only configured the routes with :locale as part of the path', () => {
-        getUrlConfig.mockImplementation(() => ({
-            locale: 'path'
-        }))
-        const CompA = () => <div>This is component A</div>
-        const CompB = () => <div>This is component B</div>
+    const env = process.env
+    beforeEach(() => {
+        jest.resetModules()
+        process.env = {...env}
+    })
 
-        const routes = [
-            {
-                path: '/',
-                component: CompA,
-                exact: true
-            },
-            {
-                path: '/category/:categoryId',
-                component: CompB,
-                exact: true
-            }
-        ]
-        const configuredRoutes = configureRoutes(routes, {ignoredRoutes: '/'})
-        expect(configuredRoutes[0].path).toEqual('/')
-        expect(configuredRoutes[1].path).toEqual('/:locale/category/:categoryId')
+    afterEach(() => {
+        process.env = env
+    })
+    const CompA = () => <div>This is component A</div>
+    const CompC = () => <div>This is component C</div>
+
+    const routes = [
+        {
+            path: '/',
+            component: CompA,
+            exact: true
+        },
+        {
+            path: '/category/:categoryId',
+            component: CompC,
+            exact: true
+        }
+    ]
+    test('should return all permutation of path including site and locales ', () => {
+        getConfig.mockImplementation(() => mockConfig)
+        const configuredRoutes = configureRoutes(routes, {ignoredRoutes: ['/']})
+        expect(configuredRoutes[configuredRoutes.length - 1].path).toEqual('/category/:categoryId')
+        expect(configuredRoutes.length).toEqual(81)
+        const paths = configuredRoutes.map((route) => route.path)
+        expect(paths).toEqual(expectedPathsResult)
+    })
+
+    test('should return the origin routes', () => {
+        getConfig.mockImplementation(() => mockConfig)
+        const configuredRoutes = configureRoutes(routes, {
+            ignoredRoutes: ['/', '/category/:categoryId']
+        })
+        expect(configuredRoutes.length).toEqual(2)
     })
 })
+
+const expectedPathsResult = [
```

Comment:
Reviewing your tests I noticed that the "id's" have priority over the "alias's". I'd think that if someone is opting to use alias's that you'd want to have them match as soon as possible, e.g. prioritizing them before the "id's".

Reply:
I get what you mean now. I've made the change [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/391/commits/2c6f8b35ba991d3dafab2d15d5d9724e614bfac5#diff-dd9d5d7fa228604e6f050e9651923dcc0dc8c03696f49eed63cbf95650543c33R28)